Lib/functools.py

```diff
@@ -388,6 +388,12 @@ def __get__(self, obj, cls):
     def __isabstractmethod__(self):
         return getattr(self.func, "__isabstractmethod__", False)
 
+# Helper functions
+
+def _unwrap_partial(func):
+    while isinstance(func, partial):
```

Comment:
@pablogsal BTW this doesn't handle the case of `partialmethod`.  Should we add that?